src/java/picard/analysis/CollectWgsMetricsFromQuerySorted.java

```diff
@@ -0,0 +1,300 @@
+        metrics.PCT_EXC_UNPAIRED = basesExcludedByPairing / (double)metrics.TOTAL_BASES;
+        metrics.PCT_EXC_BASEQ = basesExcludedByBaseq / (double)metrics.TOTAL_BASES;
+        metrics.PCT_EXC_OVERLAP = basesExcludedByOverlap / (double)metrics.TOTAL_BASES;
+        final double totalExcludedBases = basesExcludedByDupes + basesExcludedByMapq + basesExcludedByPairing + basesExcludedByBaseq + basesExcludedByOverlap;
+        metrics.PCT_EXC_TOTAL = totalExcludedBases / metrics.TOTAL_BASES;
+
+        final MetricsFile<MySeqMetrics, Integer> out = getMetricsFile();
+        out.addMetric(metrics);
+        out.write(OUTPUT);
+
+        return 0;
+    }
+
+    /**
+     * Get the count of low quality and/or softclip bases in the given read
+     *
+     * @param exclusions  the helper object
+     * @return non-negative int
+     */
+    private int getLowQualityOrSoftclipBaseCount(final BaseExclusionHelper exclusions) {
+        final byte[] quals = exclusions.read.getBaseQualities();
+
+        int badCount = exclusions.firstUnclippedBaseIndex + (quals.length - exclusions.firstTrailingClippedBaseIndex);
+        for (int i = exclusions.firstUnclippedBaseIndex; i < exclusions.firstTrailingClippedBaseIndex; i++) {
+            if (quals[i] < MINIMUM_BASE_QUALITY)
+                badCount++;
+        }
+        return badCount;
+    }
+
+    /**
+     * Get the count of overlapping bases for the given reads
+     *
+     * @param read1exclusions  the 1st read exclusions
+     * @param read2exclusions  the 2nd read exclusions
+     * @return non-negative int
+     */
+    private int getOverlappingBaseCount(final BaseExclusionHelper read1exclusions, final BaseExclusionHelper read2exclusions) {
+        // make life easy by ensuring that reads come in coordinate order
+        if ( read2exclusions.read.getAlignmentStart() < read1exclusions.read.getAlignmentStart() ) {
+            return getOverlappingBaseCount(read2exclusions, read1exclusions);
+        }
+
+        // must be overlapping
+        if ( read1exclusions.read.getAlignmentEnd() < read2exclusions.read.getAlignmentStart() ||
+                !read1exclusions.read.getReferenceIndex().equals(read2exclusions.read.getReferenceIndex()) )
+            return 0;
+
+        final byte[] read1quals = read1exclusions.read.getBaseQualities();
+        final byte[] read2quals = read2exclusions.read.getBaseQualities();
+        final int overlap = read1exclusions.read.getAlignmentEnd() - read2exclusions.read.getAlignmentStart() + 1;
+        int numHighQualityOverlappingBases = 0;
+
+        for (int i = 0; i < overlap; i++) {
+            // we only want to count it if they are both high quality (i.e. not already counted among bad bases)
+            if (read1quals[read1exclusions.firstTrailingClippedBaseIndex - overlap + i] >= MINIMUM_BASE_QUALITY && read2quals[read2exclusions.firstUnclippedBaseIndex + i] >= MINIMUM_BASE_QUALITY) {
```

Comment:
I am having a hard time understanding why this is comparing bases that are over the same genomic position....